kivy/base.py

```diff
@@ -491,9 +491,8 @@ def runTouchApp(widget=None, slave=False):
 
 def stopTouchApp():
     '''Stop the current application by leaving the main loop'''
-    if EventLoop is None:
-        return
-    if EventLoop.status != 'started':
+    if EventLoop is None or EventLoop.status != 'started':
+        Clock.schedule_once(lambda dt: stopTouchApp(), 0)
```

Comment:
@matham I dont understand the problem. In my PR calling app.stop() before eventloop is ready will schedule it, then callback will again check if eventloop is started. If it has, app will close, otherwhise the stop will be again scheduled. If the eventloop is stopped, there is no schedule, so next time user calls runTouchApp there should be no crash because callback is scheduled somewhere.

Am I not correct? Maybe I misunderstood the problem :)